index.md

```diff
@@ -1,16 +1,128 @@
+2. Install a standard container and then install the payload on top of that container instead.  For instance, `docker.io/library/python:3.12` is the official Python container, and it's built on top of Debian.  This will work if the payload isn't terribly tightly coupled to a given Linux packaging system flavor (if, for instance, the DM pipelines stack made use of RPM internally, it would be quite hard to make work on a Debian-derived system).  This may be a nonissue in practice: Open Source packages are usually portable across flavors of Linux, and even commercial ones will usually work on either RHEL or Ubuntu.  This approach also creates a larger container, but keeps the system layer fairly simple compared to using an upstream one.
+
+The first method is closer to a known quantity.  However, if a payload is not already packaged as a container, then it ends up being a variation on the second method.  The most worrying thing about this approach is that it makes accomodating new payloads harder, because significant portions of the non-payload pieces need to be rewritten for each payload.
+
+The second method means, at least theoretically, that we could keep one lab-building framework, and drop in new payloads by simply updating the single build stage that implement the installation (and runtime launch framework) of that payload.  The problem there is that there is no universal method of doing so.  However, it seems difficult to imagine that one of the following five techniques is not applicable.
+
+1. The payload is pip-installable: there's a `requirements.txt` file, the installer points `pip` at it, and the payload is installed.
+2. The payload is conda-installable: there's an environment `YAML` file, the installer points `conda` at it, and the payload is installed.
+3. The payload is supplied as a Dockerfile that constructs a container image.  The appropriate pieces of the Dockerfile can be moved into the nublado-launchable image construction (either as `COPY` and `RUN` statements, or as a shell script which does the same thing), and the payload is installed.
```

Comment:
This seems a bit cavalier about the difficulty involved in doing this. Those `COPY` and `RUN` directives may refer to a whole repository of code and potentially complex scripts, and may be assuming that they're running on a CentOS system, can install packages via `rpm`, etc. I'm fairly dubious that we want to support 3 at all.

Reply:
It is certainly harder than the other options.  I will at least push it farther down the list.

Reply:
We get to choose what we will and will not accept as potential payloads, so I think I'm going to bump it down to number four and caveat the difficulties.

In my small-n sample, other missions appear to be number one or number two, and ours is the (current) number five (bumped up to three in the revision).